### PR TITLE
added error handling for sign-up errors

### PR DESCRIPTION
**Pull request by hemmatio**
### Pull Request Title

Added Error Handling for Sign-up Errors
---

### Description

I added error messages that get flashed whenever a signup error occurs.
The signup errors are as follows:
- Missing text fields
- Username already exists in database
- Repeated password does not equal password
---

### Changes
- [x] List major changes made in this PR
- [x] Mention any new files or dependencies added
- [x] Highlight any breaking changes
- Error messages now get flashed for signup errors 
image
image
image

---

### Checklist
- [X] Code is clean and follows the project's coding standards
- [X] Documentation has been updated (if necessary)
- [X] All tests pass locally (if applicable)

---

### CODE REVIEW

File: src/interface_adapter/AccountCreation/AccountCreationPresenter.java [hunk 1]
```diff
@@ -20,16 +20,22 @@ public void setPassView(AccountCreationOutputData response) {
...
-//        accountCreationViewModel.setState(accountCreationState);
-//        accountCreationViewModel.firePropertyChanged(error);
-        System.out.println("Error");
+        AccountCreationState accountCreationState = accountCreationViewModel.getState();
+        accountCreationState.setErrorCode(error);
+        if (error.equals("unmatched_passwords")) {
+            accountCreationState.setErrorCode("Passwords do not match.");
+        }
+        if (error.equals("missing_fields")) {
+            accountCreationState.setErrorCode("Please fill in all fields.");
```
Comment (Lakssito) [inline src/interface_adapter/AccountCreation/AccountCreationPresenter.java:34]: setErrorCode should have another variable name. This one is not descriptive enough

**Review by Lakssito — CHANGES_REQUESTED** [review]
Please read annotated comments and change required variable names.

File: src/interface_adapter/AccountCreation/AccountCreationPresenter.java (hunk 1 above)
Comment (hemmatio) [reply src/interface_adapter/AccountCreation/AccountCreationPresenter.java:34]: You're right, I'll refactor this variable name to be more indicative of what it serves. Thanks for the feedback and review.

**Review by Lakssito — APPROVED** [review]
Nice changes as requested, I approve the pull request.

**Pull request merged**